dojson/contrib/to_marc21/fields/bd25x28x.py

```diff
@@ -285,45 +285,88 @@ def reverse_production_publication_distribution_manufacture_and_copyright_notice
 def reverse_address(self, key, value):
     """Reverse - Address."""
     indicator_map1 = {
-        "No level specified": "_",
         "Primary": "1",
         "Secondary": "2"}
+
+    indicator_map2 = {
+        "Mailing": "0",
+        "Type specified in subfield $i": "7"}
+
+    field_map = {
+        'address': 'a',
+        'city': 'b',
+        'state_or_province': 'c',
+        'country': 'd',
+        'postal_code': 'e',
+        'terms_preceding_attention_name': 'f',
+        'attention_name': 'g',
+        'attention_position': 'h',
+        'type_of_address': 'i',
+        'specialized_telephone_number': 'j',
+        'telephone_number': 'k',
+        'fax_number': 'l',
+        'electronic_mail_address': 'm',
+        'tdd_or_tty_number': 'n',
+        'contact_person': 'p',
+        'title_of_contact_person': 'q',
+        'hours': 'r',
+        'public_note': 'z',
+        'relator_code': '4',
+        'linkage': '6',
+        'field_link_and_sequence_number': '8',
+    }
+
+    order = utils.map_order(field_map, value)
+
+    if key[3] in indicator_map1:
+        order.append('level')
+
+    if key[4] in indicator_map2:
+        order.append('type_of_address')
+
+    if indicator_map2.get(value.get('type_of_address'), '7') != '7':
+        order.remove('i')
+
     return {
+        '__order__': tuple(order) if len(order) else None,
         'a': utils.reverse_force_list(
             value.get('address')
         ),
-        'c': value.get('state_or_province'),
         'b': value.get('city'),
-        'e': value.get('postal_code'),
+        'c': value.get('state_or_province'),
         'd': value.get('country'),
-        'g': value.get('attention_name'),
+        'e': value.get('postal_code'),
         'f': value.get('terms_preceding_attention_name'),
-        'i': value.get('type_of_address'),
+        'g': value.get('attention_name'),
         'h': value.get('attention_position'),
-        'k': utils.reverse_force_list(
-            value.get('telephone_number')
-        ),
+        'i': value.get('type_of_address'),
```

Comment:
remove this duplicate 'i' key, it is being set at the bottom of this return correctly